### PR TITLE
feat: onetrust for cloud mode events

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

If OneTrust is enabled and consent has been given, fetch the deny list and pass it down as context property.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: src/utils/utils.js [hunk 1]
```diff
@@ -769,6 +769,26 @@ const getStringId = (id) => {
...
+/**
+ * Function to return the state of consent management based on config passed in load options 
+ * @param {Object} cookieConsentOptions 
+ * @returns 
+ */
+const fetchCookieConsentState = (cookieConsentOptions) => {
+  let isEnabled = false;
+  // eslint-disable-next-line consistent-return
+  Object.keys(cookieConsentOptions).forEach((e) => {
+    if (
```
Comment (bardisg) [inline src/utils/utils.js:781]: it is a good practice on conditonals to assign logical results in a variable thatis selfdecraring of what the calculated value is before using it. Like:

const isSupportedAndEnabled = SUPPORTED_CONSENT_MANAGERS.includes(e) && 
      typeof(cookieConsentOptions[e].enabled) === 'boolean' && 
      cookieConsentOptions[e].enabled === true;

if(isSupportedAndEnabled) {

easier to read the code, split the value calculation if needed in separate util functions, reuse the result, amend the logic

File: src/features/core/cookieConsent/OneTrust/browser.js [hunk 2]
```diff
@@ -22,13 +22,16 @@ class OneTrust {
 
     const oneTrustAllGroupsInfo = window.OneTrust.GetDomainData().Groups;
     this.userSetConsentGroupNames = [];
```
Comment (bardisg) [inline src/features/core/cookieConsent/OneTrust/browser.js:24]: Can we rename this to userProvidedConsentGroupIds please?

File: src/core/analytics.js [hunk 3]
```diff
@@ -838,6 +842,12 @@ class Analytics {
       } catch (e) {
         handleError(e);
       }
+      // If cookie consent is enabled attach the denied consent group Ids to the context
+      if (fetchCookieConsentState(this.cookieConsentOptions)) {
```
Comment (bardisg) [inline src/core/analytics.js:846]: can we make use that context.consentManagement cannot be overriden by cstom apiOptions like the library details please?

File: __tests__/oneTrustCookieConsent.test.js [hunk 4]
```diff
@@ -4,12 +4,18 @@ window.OneTrust = {
   GetDomainData: jest.fn(() => ({
     Groups: [
       { CustomGroupId: 'C0001', GroupName: 'Functional Cookies' },
+      { CustomGroupId: 'C0002', GroupName: 'Performance Cookies' },
```
Comment (bardisg) [inline __tests__/oneTrustCookieConsent.test.js:7]: can we also add a test in sanity sute for this?

- add a function in trigger that sets dummy concent ids in window for a test case in track
- clear the value afterwards adding a new resetWindow method in the testbook.js that is called after result diff check

**Review by bardisg — DISMISSED** [review]

File: src/utils/utils.js (hunk 1 above)
Comment (MoumitaM) [reply src/utils/utils.js]: Adressed

File: src/core/analytics.js (hunk 3 above)
Comment (MoumitaM) [reply src/core/analytics.js:847]: Adressed

**Review by bardisg — APPROVED** [review]

**Comment by bardisg** [comment]
This should be merged in staging and releashed only after we test end E2E with server version 1.6.0 that supports the feature

**Review by bardisg — APPROVED** [review]

**Pull request merged**